src/server/IoDispatchers.cpp

```diff
@@ -299,6 +297,9 @@ PCONSOLE_API_MSG IoDispatchers::ConsoleHandleConnectionRequest(_In_ PCONSOLE_API
                 gci.ProcessHandleList.FreeProcessData(ProcessData);
             }
         }
+
+        // FreeProcessData() above requires the console to be locked.
+        UnlockConsole();

```

Comment:
For reviewers, here's the original code again:
https://github.com/microsoft/terminal/blob/40e3deae4640d0f10ac70e1cdb68adb900f4ba1c/src/server/IoDispatchers.cpp#L441-L454

This is the fail-fast:
https://github.com/microsoft/terminal/blob/d0d42c4ba1196148c973f5fbaaa5516a76c8507b/src/server/ProcessList.cpp#L86-L88

Reply:
FYI the stack trace is:
```
0:000> kc
 # Call Site
00 KERNELBASE!RaiseFailFastException
01 conhost!wil::details::WilDynamicLoadRaiseFailFastException
02 conhost!wil::details::WilRaiseFailFastException
03 conhost!wil::details::WilFailFast
04 conhost!wil::details::ReportFailure_NoReturn<3>
05 conhost!wil::details::ReportFailure_Base<3,0>
06 conhost!wil::details::ReportFailure_Hr<3>
07 conhost!wil::details::in1diag3::_FailFast_Unexpected
08 conhost!wil::details::in1diag3::FailFast_If
09 conhost!ConsoleProcessList::FreeProcessData
0a conhost!<lambda_f86d5fd29b665fa3f430c14358e0681a>::operator()
0b conhost!wil::details::lambda_call<<lambda_f86d5fd29b665fa3f430c14358e0681a> >::reset
0c conhost!wil::details::lambda_call<<lambda_f86d5fd29b665fa3f430c14358e0681a> >::~lambda_call<<lambda_f86d5fd29b665fa3f430c14358e0681a> >
0d conhost!IoDispatchers::ConsoleHandleConnectionRequest
0e conhost!IoSorter::ServiceIoOperation
0f conhost!ConsoleIoThread
10 kernel32!BaseThreadInitThunk
11 ntdll!RtlUserThreadStart
```